### PR TITLE
tools: improve api bench

### LINKED ISSUE #7167: Tracking issue for self-adaptive rate limit in self protection
## Development Task

Summary:
Refer to https://github.com/tikv/pd/issues/4373.
We have implemented gRPC rate-limit and HTTP rate-limit by manually setting the rate-limiting parameters.
But it's hard to set a speed limit. Because the carrying capacity of different clusters is different, and the carrying capacity of different interfaces in different loads is also different.

So we should provide a mechanism to set rate limit adaptively.

Here are some references for applying the tcp BBR algorithm to system traffic limiting scenarios.
https://github.com/alibaba/Sentinel/wiki/%E7%B3%BB%E7%BB%9F%E8%87%AA%E9%80%82%E5%BA%94%E9%99%90%E6%B5%81
https://github.com/go-kratos/aegis/blob/main/ratelimit/bbr/bbr.go

Tasks：
- [ ] Implement BBR algorithm
  - [ ] CPU stats
  - [ ] Sliding Time window
  - [ ] BBR
- [ ] Apply BBR in PD
  - [ ] Which APIs to choose and classify
  - [ ] Config

### PR DESCRIPTION
**Pull request by CabinfeverB**
### What problem does this PR solve?

Issue Number: ref #7167 

### What is changed and how does it work?
It will help test active-follower and rate limit

```commit-message
```

### Check List

Tests

- Manual test (add detailed scripts or steps below)

Code changes

- Has the configuration change
- Has HTTP API interfaces changed (Don't forget to [add the declarative for the new API](https://github.com/tikv/pd/blob/master/docs/development.md#updating-api-documentation))
- Has persistent data change

Side effects

- Possible performance regression
- Increased code complexity
- Breaking backward compatibility

Related changes

- PR to update [`pingcap/docs`](https://github.com/pingcap/docs)/[`pingcap/docs-cn`](https://github.com/pingcap/docs-cn):
- PR to update [`pingcap/tiup`](https://github.com/pingcap/tiup):
- Need to cherry-pick to the release branch

### Release note

```release-note
None.
```

### CODE REVIEW

File: tools/pd-api-bench/cases/cases.go [hunk 1]
```diff
@@ -254,6 +260,44 @@ func (c *getRegion) Unary(ctx context.Context, cli pd.Client) error {
...
+	return &storeHeartbeat{
+		baseCase: &baseCase{
+			name:  "StoreHeartbeat",
+			qps:   10000,
+			burst: 1,
+		},
+	}
+}
+
+func (c *storeHeartbeat) Unary(ctx context.Context, cli pd.Client) error {
```
Comment (bufferflies) [inline tools/pd-api-bench/cases/cases.go:277]: how about rename `SendStoreHeartbeat`?

Comment (CabinfeverB) [reply tools/pd-api-bench/cases/cases.go:277]: `Unary` impls GRPCCase

File: tools/pd-api-bench/main.go [hunk 2]
```diff
@@ -47,6 +49,10 @@ var (
 	qps   = flag.Int64("qps", 1000, "qps")
 	burst = flag.Int64("burst", 1, "burst")
 
+	wait = flag.Bool("wait", true, "wait for a round")
```
Comment (HuSharp) [inline tools/pd-api-bench/main.go:52]: pls update `readme.md` as well :)

File: tools/pd-api-bench/main.go [hunk 3]
```diff
@@ -307,8 +313,13 @@ func handleHTTPCase(ctx context.Context, hcase cases.HTTPCase, httpClis []pdHttp
 					for i := int64(0); i < burst; i++ {
 						cntMu.Lock()
 						startCnt++
-						if *metrics && startCnt%1000 == 0 {
-							log.Printf("case http %s has done query %d", hcase.Name(), startCnt)
+						if *metrics > 0 {
+							cntMu.Lock()
```
Comment (bufferflies) [inline tools/pd-api-bench/main.go:317]: It may bring the dead lock, golang doesn't support the Reentrant lock.

**Pull request closed without merging**